From ca19cb36337bcb9feb6bd380fe9048535fa04c71 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Sun, 12 Aug 2018 15:47:41 +0800
Subject: [PATCH 4384/5242] MLK-19158-4 drm/imx: lcdif: remove bus format
 check from PLANE's check

commit  c23dc94444cfac2ad3edebbfbf46316d177290e0 from
https://source.codeaurora.org/external/imx/linux-imx.git

Since the LCDIF output data width can be different from the data
width of input pixel data, so the bus format check in the plane's
atomic check is not correct anymore, and need to be removed.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit 2245702e7905fa7b75aec92fdbb9ffeb33bdb6de)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/lcdif/lcdif-plane.c |   13 -------------
 1 file changed, 13 deletions(-)

diff --git a/drivers/gpu/drm/imx/lcdif/lcdif-plane.c b/drivers/gpu/drm/imx/lcdif/lcdif-plane.c
index 6662ba5..a29fd3d 100644
--- a/drivers/gpu/drm/imx/lcdif/lcdif-plane.c
+++ b/drivers/gpu/drm/imx/lcdif/lcdif-plane.c
@@ -45,16 +45,12 @@ static int lcdif_plane_atomic_check(struct drm_plane *plane,
 				    struct drm_plane_state *plane_state)
 {
 	int ret;
-	uint32_t bus_fmt;
-	struct lcdif_plane *lcdif_plane = to_lcdif_plane(plane);
-	struct lcdif_soc *lcdif = lcdif_plane->lcdif;
 	struct drm_plane_state *old_state = plane->state;
 	struct drm_framebuffer *fb = plane_state->fb;
 	struct drm_framebuffer *old_fb = old_state->fb;
 	struct drm_crtc_state *crtc_state;
 	struct drm_display_mode *mode;
 	struct drm_rect clip = { 0 };
-	unsigned int flags;
 
 	/* 'fb' should also be NULL which has been checked in
 	 * the core sanity check function 'drm_atomic_plane_check()'
@@ -72,15 +68,6 @@ static int lcdif_plane_atomic_check(struct drm_plane *plane,
 							plane_state->crtc);
 	mode = &crtc_state->adjusted_mode;
 
-	bus_fmt = lcdif_get_bus_fmt_from_pix_fmt(lcdif, fb->format->format);
-	if (bus_fmt < 0)
-		return -EINVAL;
-
-	/* check fb pixel format matches bus format */
-	flags = mode->private_flags & 0xffff;
-	if (flags != bus_fmt)
-		return -EINVAL;
-
 	clip.x2 = mode->hdisplay;
 	clip.y2 = mode->vdisplay;
 
-- 
1.7.9.5

